src/redux/books/booksSlice.js

```diff
@@ -1,17 +1,94 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
 
-const initialState = [];
+const API_ENDPOINT = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/px0dcsoOe2YK23gCLe5s/books';
+
+export const fetchBooks = createAsyncThunk('book/fetchBooks', async () => {
+  const response = await axios.get(API_ENDPOINT);
+  try {
+    return response.data;
+  } catch (error) {
+    return new Error(error.message);
+  }
+});
+
+export const initialState = {
+  books: [],
+  status: 'idle',
+  error: null,
+};
+
+// export const addBook = createAsyncThunk('book/addBook', async (book) => {
+//   try {
+//     const response = await axios.post(API_ENDPOINT, book);
+//     return response.data;
+//   } catch (error) {
+//     throw new Error(error.message);
+//   }
+// });
+
+// export const removeBook = createAsyncThunk('book/removeBook', async (item_id) => {
+//   try {
+//     await axios.delete(`${API_ENDPOINT}/${item_id}`);
+//     return item_id;
+//   } catch (error) {
+//     throw new Error(error.message);
+//   }
+// });
 
 const booksSlice = createSlice({
   name: 'books',
   initialState,
   reducers: {
-    addBook: (state, action) => {
-      state.push(action.payload);
+
+    addBook: (state, { payload }) => {
+      state.books.push(payload);
     },
-    removeBook: (state, action) => state.filter((book) => book.item_id !== action.payload),
+
+    removeBook: (state, { payload }) => {
+      state.books = state.books.filter((book) => book.item_id !== payload);
+    },
+
```

Comment:
- [ ] Great job working on the required changes so far but I think you still have a small problem when I try to delete a book so whenever I remove a book from the list I could see allthe books are removed as well but I think it shouldn't be like that so please fix that problem to just whenever the user want to remove a specific book it removed the book  not all list 🔥💯⭐🚀


https://github.com/tomdan-ai/Bookstore/assets/58553711/7c2f17e9-be5c-47d7-a07b-2a09808fe511